src/System.Management.Automation/commands/utility/FormatAndOutput/out-console/OutConsole.cs

```diff
@@ -143,15 +148,30 @@ protected override void EndProcessing()
             }
 
             base.EndProcessing();
+        }
 
-            if (Transcript)
+        /// <summary>
+        /// Revert transcription state on Dispose
+        /// </summary>
+        protected override void InternalDispose()
+        {
+            if (_disposed)
+            {
+                return;
+            }
+
+            if (_transcriptStateChanged)
             {
                 Host.UI.TranscribeOnly = _savedTranscribeOnly;
+                _transcriptStateChanged = false;
             }
+            _disposed = true;
         }
 
+        private bool _disposed = false;
         private ArrayList _outVarResults = null;
         private bool _savedTranscribeOnly = false;
+        private static bool _transcriptStateChanged = false;
```

Comment:
You define `_transcriptStateChanged` as `static`, so it affect entire `AppDomain`. That might be not good when multiple PowerShell `Runspace`s share same `AppDomain`.